### DISCUSSION TITLE
NextJS: ES Modules is not supported

### DISCUSSION THREAD

**Discussion by stephencweiss** [vercel/community#588] [Help] [answered]
I'm having a heck of a time getting NextJS to process my markdown blog.

I've found a few different blogs / posts / etc. suggesting _how_ to do this with `remark` and/or `unifed`. It works locally just fine, but I'm getting a variety of errors when I actually try to push the code and deploy with Vercel.

Latest attempt:
```ts
import { unified } from "unified";
import mdParse from "remark-parse";
import prism from "remark-prism";
import remarkGfm from "remark-gfm";
import remarkRehype from "remark-rehype";
import remarkStringify from "remark-stringify";

export async function markdownToHtml(content: string) {
  return await unified()
    .use(mdParse)
    .use(prism)
    .use(remarkGfm)
    .use(remarkRehype)
    .use(remarkStringify)
    .process(content);
}
```

```ts
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
};
const withTM = require("next-transpile-modules")(["unifed"]);
module.exports = withTM(nextConfig);
```

```sh
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /vercel/path0/node_modules/unified/index.js
--
18:05:05.421 | require() of ES modules is not supported.
18:05:05.421 | require() of /vercel/path0/node_modules/unified/index.js from /vercel/path0/utils/serverUtils/markdownToHtml.ts is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
18:05:05.421 | Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /vercel/path0/node_modules/unified/package.json.
18:05:05.421 |  
18:05:05.422 | at new NodeError (internal/errors.js:322:7)
18:05:05.422 | at Module._extensions..js (internal/modules/cjs/loader.js:1102:13)
18:05:05.422 | at Object.require.extensions.<computed> [as .js] (/vercel/path0/node_modules/ts-node/src/index.ts:1449:43)
18:05:05.422 | at Module.load (internal/modules/cjs/loader.js:950:32)
18:05:05.422 | at Function.Module._load (internal/modules/cjs/loader.js:790:12)
18:05:05.422 | at Module.require (internal/modules/cjs/loader.js:974:19)
18:05:05.422 | at require (internal/modules/cjs/helpers.js:101:18)
18:05:05.422 | at Object.<anonymous> (/vercel/path0/utils/serverUtils/markdownToHtml.ts:1:1)
18:05:05.422 | at Module._compile (internal/modules/cjs/loader.js:1085:14)
18:05:05.422 | at Module.m._compile (/vercel/path0/node_modules/ts-node/src/index.ts:1459:23) {
```

I also tried this with:
```ts
export default async function markdownToHtml(markdown) {
  const result = await remark()
    // https://github.com/sergioramos/remark-prism/issues/265
    .use(html, { sanitize: false })
    .use(prism)
    .process(markdown);
  return result.toString();
}
```

This code comes from https://github.com/leerob/nextjs-prism-markdown/blob/main/lib/markdown.js. 

The only (?) difference I can find in how it's used in the example with my code is that @leerob puts the function inside `getStaticProps` and I put mine inside a call to `getServerSideProps` because I am using iron-session (https://github.com/vvo/iron-session#nextjs-withironsessionapiroutehandler-ironoptions--req-nextapirequest-res-nextapiresponse--ironoptions--promiseironoptions).

This is one in a long list of differences I've had where things work "perfectly" when running `npm run dev` but fail catastrophically when deployed. 

I'm at a loss. Any ideas on what I'm doing wrong? If I'm not mistaken, I'm using the same versions of the packages as demonstrated in `nextjs-prism-markdown`.

**Comment by swarnava** [comment, marked as answer]
Hi,

Please report it here: https://github.com/vercel/next.js/issues

**Reply by stephencweiss** [reply]
@swarnava - I will, but I'm curious: is this a nextjs issue or a vercel issue?

**Reply by swarnava** [reply]
Hi, at first glance, it sounds like a Next.js issue or issue with the way you have implemented. The respective team should be able to comment. Thanks!

**Comment by stephencweiss** [comment]
Thank you! I ask because I started in the NextJS discord server and was pointed her 😄

I’ll try to pull together a small example for the bug this week. Unfortunately, part of the complication is that some of the content is private and so I’m using submodules.
